File: C++/Baekjoon/Problem1074.cpp
#include <iostream>
#include <cmath>

using namespace std;

int N, r, c, res;

void sol(int x, int y, int sz) {
    if (c == x && r == y) {
        cout << res;
        return;
    }
    if (c < x + sz && r < y + sz && c >= x && r >= y) {
        sol(x, y, sz / 2);
        sol(x + sz / 2, y, sz / 2);
        sol(x, y + sz / 2, sz / 2);
        sol(x + sz / 2, y + sz / 2, sz / 2);
    } else res += sz * sz;
}

int main() {
    cin >> N >> r >> c;
    sol(0, 0, pow(2, N));
}

File: Python/Baekjoon/Problem1074.py
def sol(x, y, sz):
    global res
    if c == x and r == y:
        print(res)
        return
    if c < x + sz and r < y + sz and c >= x and r >= y:
        sol(x, y, sz // 2)
        sol(x + sz // 2, y, sz // 2)
        sol(x, y + sz // 2, sz // 2)
        sol(x + sz // 2, y + sz // 2, sz // 2)
    else:
        res = res + sz * sz


N, r, c = map(int, input().split())
res = 0
sol(0, 0, 2 ** N)
